background.js

```diff
@@ -1,22 +1,53 @@
 const pluginFlag = "pluginFlag"
+const translatedFlag = "pluginFlag"
 var shouldTranslate = false
 let initalRequestMethod = "shouldTranslate"
 let updateRequestMethod = "updateShouldTranslate"
 let reloadRequestMethod = "reloadShouldTranslate"
-
+let startTranslateRequestMethod = "startTranslate"
+var currentTranslatingPage = [];
+var currentTabID = 0
+var previousTabID = 0
 
 chrome.storage.local.get(pluginFlag, (result) => {
     shouldTranslate = result.pluginFlag || false
-  });
+});
 
 chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
     if (request.type === updateRequestMethod) {
         shouldTranslate = request.data;
-        chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
-          chrome.tabs.reload(tabs[0].id);
+        currentTranslatingPage = []
+        currentTabID = 0
+        console.log("clear" + " Current" + currentTabID)
+        chrome.tabs.query({}, function (tabs) {
+           let running = []
+           for (let tab of tabs) {
+               if (tab.url.includes("developer.apple.com")) {
+                   running.push(chrome.tabs.reload(tab.id))
+               }
+           }
+
+           Promise.allSettled(running).then()
+
+           chrome.tabs.query({active: true, currentWindow: true}, function (tabs) {
```

Comment:
Can we unify these two `chrome.tabs.query` functions?  Distinguish the by current page URL or something else.